src/renderer/screens/settings.tsx

```diff
@@ -0,0 +1,157 @@
+import {
+    Box,
+    Flex,
+    FormLabel,
+    Input,
+    Select
+  } from '@chakra-ui/react';
+  import { useState } from 'react';
+  import Container from 'renderer/components/container';
+  import { adjustRam } from 'renderer/services/ram.service';
+  import { adjustNick } from 'renderer/services/nickname.service';
+
+  import playerData from '../../common/player.json';
+  import ramData from '../../common/ram.json';
+  // Placeholder values for RAM adjustment
+  var plRamMin = '';
+  var plRamMax = '';
+  
+  interface StateNick {
+    value: string;
+    setValue: React.Dispatch<React.SetStateAction<string>>;
+  }
+  interface StateRam {
+    minval: string;
+    maxval: string;
+    setRammn: React.Dispatch<React.SetStateAction<string>>;
+    setRammx: React.Dispatch<React.SetStateAction<string>>;
+    selects: number;
+    setSelects: React.Dispatch<React.SetStateAction<number>>;
+  }
+  
+  // Valid chars in nickname
+  const VALID_NAME_CHARS = /^[a-zA-Z0-9-_]+$/;
+
+  // Check if nickname is valid and apply the changes
+  function checkNick(nick: string) {
+    for (let i = 0; i < nick.length; i += 1) {
+      if (!nick.match(VALID_NAME_CHARS) || nick.length < 3 ||  nick.length > 16) {
+        return true;
+      }
+      else {
+        // Adjust nickname
+        adjustNick(nick);
```

Comment:
I liked this part of name validation. But I don't think illegal names should be adjusted or banned. We could just place a warning.